Sources/Command/Command.swift

```diff
@@ -238,8 +231,10 @@ public struct CommandRunner: CommandRunning, Sendable {
         }
     }
 
-    func lookupExecutable(firstArgument: String?) throws -> URL? {
-        guard let firstArgument else { return nil }
+    func lookupExecutable(firstArgument: String?) throws -> URL {
+        guard let firstArgument else {
+            throw CommandError.executableNotFound("(nil)")
```

Comment:
What about adding a new error case instead of reusing the existing one with `(nil)`, which is hard to tell what it means?

```suggestion
            throw CommandError.missingExecutableName
```